runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/budget/EvenGetWorkBudgetDistributor.java

```diff
@@ -67,17 +70,18 @@ public <T extends GetWorkBudgetSpender> void distributeBudget(
       GetWorkBudgetSpender getWorkBudgetSpender = streamAndDesiredBudget.getKey();
       GetWorkBudget desired = streamAndDesiredBudget.getValue();
       GetWorkBudget remaining = getWorkBudgetSpender.remainingBudget();
-      if (isBelowFiftyPercentOfTarget(remaining, desired)) {
+      if (isBelowFiftyPercentOfTarget(remaining, desired) && isActiveWorkBudgetAware) {
```

Comment:
this is unclear to me.  It seems like in one case the adjustment is addtiive and another is resetting

Wouldn't we want to only increase by the desired-remaining in either case?

Reply:
whether its overriding or additive is depending on the internal implementation of `GetWorkBudgetSpender.adjustBudget()`

based on the current implementation that this is reflecting, we do not account for remaining budget here it is handled internally in adjustBudget.